src/Text/Mustache/Compile.hs

```diff
@@ -143,6 +148,45 @@ getFile (templateDir : xs) fp =
     filePath = templateDir </> fp
 
 
+-- |
+-- Compile a mustache 'Template' at compile time. Usage:
+--
+-- > {-# LANGUAGE QuasiQuotes #-}
+-- > import Text.Mustache.Compile (mustache)
+-- >
+-- > foo :: Template
+-- > foo = [mustache|This is my inline {{ template }} created at compile time|]
+--
+-- Partials are not supported in compile time templates.
+
+mustache ∷ QuasiQuoter
+mustache = QuasiQuoter {quoteExp = \unprocessedTemplate → do
+  l ← location
+  compileTemplateTH (fileAndLine l) unprocessedTemplate }
+
+fileAndLine ∷ Loc → String
+fileAndLine loc = loc_filename loc ++ ":" ++ (show ∘ fst ∘ loc_start $ loc)
+
+compileTemplateTH ∷ String → String → Q Exp
+compileTemplateTH filename unprocessed =
+  either (fail ∘ ("Parse error in mustache template: " ++) ∘ show) THS.lift $ compileTemplate filename (pack unprocessed)
+
+-- |
+-- Compile a mustache 'Template' at compile time. Usage:
+--
+-- > {-# LANGUAGE TemplateHaskell #-}
+-- > import Text.Mustache.Compile (embedTemplate)
+-- >
+-- > foo :: Template
+-- > foo = $(embedTemplate "dir/file.mustache")
+--
+-- Partials are not supported in compile time templates.
+
+embedTemplate :: FilePath → Q Exp
+embedTemplate filePath = do
+  stringFile ← THS.runIO $ readFile filePath
```

Comment:
[`qAddDependentFile`](https://hackage.haskell.org/package/template-haskell-2.10.0.0/docs/Language-Haskell-TH-Syntax.html#v:qAddDependentFile) `>> readFile filePath`? So that whenever the file changes, the template is automatically recompiled.